flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinFlowPluginExtension.kt

```diff
@@ -143,7 +143,7 @@ public open class VaadinFlowPluginExtension(project: Project) {
     /**
      * The folder where flow will put TS API files for client projects.
      */
-    public var generatedTsFolder: File = File(project.projectDir, "frontend/generated")
+    public var generatedTsFolder: File? = null
```

Comment:
Why defer the initialization? It is only dependent on the `frontendDirectory` which has already been initialized and has not changed. 
```suggestion
    public var generatedTsFolder: File = File(frontendDirectory, FrontendUtils.GENERATED)
```